src/Sema.zig

```diff
@@ -28701,15 +28701,34 @@ fn elemPtrArray(
     block: *Block,
     src: LazySrcLoc,
     array_ptr_src: LazySrcLoc,
-    array_ptr: Air.Inst.Ref,
+    array_ptr_orig: Air.Inst.Ref,
     elem_index_src: LazySrcLoc,
     elem_index: Air.Inst.Ref,
     init: bool,
     oob_safety: bool,
 ) CompileError!Air.Inst.Ref {
     const pt = sema.pt;
     const zcu = pt.zcu;
-    const array_ptr_ty = sema.typeOf(array_ptr);
+    const array_ptr_ty_orig = sema.typeOf(array_ptr_orig);
+
+    var ptr_info = array_ptr_ty_orig.ptrInfo(zcu);
+    const array_ptr, const array_ptr_ty = if (ptr_info.flags.size == .c) blk_outer: {
+        const array_ptr_orig_inst = array_ptr_orig.toIndex().?;
+        const air_tags = sema.air_instructions.items(.tag);
```

Comment:
Why are you looking at the instruction instead of the type? Did checking the type not work?